ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -184,14 +184,19 @@ let set_diagnostics rpc doc =
     async (fun () -> Diagnostics.send state.diagnostics (`One uri))
   | Reason
   | Ocaml ->
-    async (fun () ->
-        let* diagnostics =
-          let command =
-            Query_protocol.Errors
-              { lexing = true; parsing = true; typing = true }
-          in
-          Document.with_pipeline_exn doc (fun pipeline ->
-              let errors = Query_commands.dispatch pipeline command in
+    let send () =
+      let* diagnostics =
+        let command =
+          Query_protocol.Errors { lexing = true; parsing = true; typing = true }
+        in
+        Document.with_pipeline_exn doc (fun pipeline ->
+            match Query_commands.dispatch pipeline command with
+            | exception Extend_main.Handshake.Error error ->
+              let message =
+                sprintf "%s.\nHint: install merlin-extend reason" error
```

Comment:
What's "merlin-extend reason"? Something like "install `merlin-extend` package" or something in that spirit would be more understandable I think.